### ISSUE TITLE
[BUG] `PartialServerEndpoint` is losing info when generating `OpenApi`

### ISSUE THREAD

**Issue by tdroxler** [softwaremill/tapir#668]
Tapir version: v0.16.2 

**Describe the bug**
When using `Endpoint.serverLogicForCurrent`, the input is set as empty as well as the `I` type to `Unit`: 

https://github.com/softwaremill/tapir/blob/808f3ef8b226beb684657f6f105b194f453ac226/core/src/main/scala/sttp/tapir/Endpoint.scala#L315-L319

This implementation works well  `toRoute` the endpoint, but then the `openapi.yaml` file miss the information of the first input.

**How to reproduce?**
Using [the same example as in the wiki](https://tapir.softwaremill.com/en/latest/server/logic.html?highlight=andThen#defining-an-extendable-base-endpoint-with-partial-server-logic)  but using a `query` for the `User` to makes it even more obvious:

```
val secureEndpoint: PartialServerEndpoint[User, Unit, Int, Unit, Nothing, Future] =
  endpoint
    .in("secure")
    .in(query[String]("token"))
    .errorOut(plainBody[Int])
    .serverLogicForCurrent(auth)

val secureHello: PartialServerEndpoint[User, String, Int, String, Nothing, Future] =
   secureEndpoint.get
    .in("hello")
    .in(query[String]("salutation"))
    .out(stringBody)
```
then if we generate the openapi we get: 
![maim-1594987185](https://user-images.githubusercontent.com/2979182/87784022-da544880-c835-11ea-9092-333ba41b1fa1.png)

We see that we miss the `secure` path and the `token` info.

**Additional information**
I think this could be solved by using the same approach as in the `ServerEndpointInParts`, where the input is split 

https://github.com/softwaremill/tapir/blob/6895cd50c5bb5181d1a31fbcdd4de95473d76995/core/src/main/scala/sttp/tapir/server/ServerEndpointInParts.scala#L37

Thank you very much

Cheers,

Thomas

**Comment by adamw** [comment]
I've tried reproducing the problem but the generated yaml I get using:

```
val pe: PartialServerEndpoint[String, Unit, Int, Unit, Nothing, Future] =
      endpoint
        .in("secure")
        .in(query[String]("token"))
        .errorOut(plainBody[Int])
        .serverLogicForCurrent(_ => Future.successful(Right(""): Either[Int, String]))

    val e = pe.get
      .in("hello")
      .in(query[String]("salutation"))
      .out(stringBody)
      .serverLogic(_ => Future.successful(Right("ok")))

    val actualYaml = List(e).toOpenAPI(Info("Partial", "1.0")).toYaml
```

contains both parameters:

```
      parameters:
      - name: token
        in: query
        required: true
        schema:
          type: string
      - name: salutation
        in: query
        required: true
        schema:
          type: string
```

Can you check the generated yaml? Also, how do you convert endpoints to yaml?

**Comment by tdroxler** [comment]
ok I got it, I'm currently doing for all my other endpoints :
```
  val getUser: Endpoint[Unit, Int, String, Nothing] =
    endpoint.get
      .in("user")
      .out(stringBody)
      .errorOut(plainBody[Int])
  
  val docs: OpenAPI = List(getUser).toOpenAPI("Test", "1.0")
```
So I generate my openapi with a list of `Endpoint`, not `ServerEndpoint` I found it more elegant. But it means that I can't do
```
val docs: OpenAPI = List(getUser, secureHello).toOpenAPI("Test", "1.0")
```
So the workaround I did and which causes my problem is to have `secureHello.endpoint` to generate the openapi and the loss of information is there.

Don't you think we should be able to generate the doc from a `PartialServerEndpoint` without having to `serverLogic` it?

I'll generate the openapi with the logic to make it work.

**Comment by adamw** [comment]
But if I do `secureEndpoint.endpoint.toOpenAPI(...)` I also get all of there parameters. So the problem might be somewhere else :)

**Comment by adamw** [comment]
Ah wait ... you're calling .endpoint on the partial one. Then it breaks. Ok, so at least now I can reproduce, next let's think how to fix this :)

**Comment by adamw** [comment]
Fixed in 0.16.5

**Comment by tdroxler** [comment]
thx a lot
